File: benchmarks/python/04_characteristics.py
"""Benchmark 04: Cohort Characteristics."""

from helpers import Timer, connect_cdm, save_result, save_timing

from omopy.characteristics import summarise_characteristics
from omopy.connector import generate_concept_cohort_set
from omopy.generics import Codelist

print("=== 04: Cohort Characteristics ===")
t = Timer()
cdm = connect_cdm()

cdm = generate_concept_cohort_set(
    cdm, Codelist({"coronary_artery": [317576]}), name="target_cohort"
)
cohort = cdm.cohort_tables["target_cohort"]

result = summarise_characteristics(cohort)
save_result(result.data, "04_characteristics")
save_timing("04_characteristics", t.elapsed())
print(f"Done in {t.elapsed():.2f} seconds")

File: benchmarks/r/04_characteristics.R
# Benchmark 04: Cohort Characteristics
# R equivalent: CohortCharacteristics::summariseCharacteristics()

source("benchmarks/r/00_helpers.R")
library(CohortCharacteristics)
cat("=== 04: Cohort Characteristics ===\n")
t0 <- proc.time()

con <- dbConnect(duckdb(), dbdir = DB_PATH)
cdm <- cdmFromCon(con = con, cdmSchema = "main", writeSchema = "main", cdmName = "synthea_1k")

cdm <- generateConceptCohortSet(
  cdm = cdm,
  conceptSet = list(coronary_artery = 317576),
  name = "target_cohort"
)

result <- summariseCharacteristics(cdm$target_cohort)
save_result(as.data.frame(result), "04_characteristics")
elapsed <- (proc.time() - t0)[["elapsed"]]
save_timing("04_characteristics", elapsed)
cdmDisconnect(cdm)
cat("Done in", elapsed, "seconds\n")